### PR TITLE
Remove test publishing

### PR DESCRIPTION
**Pull request by lanej0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

1. Swap the HPOS and non-HPOS testing workflows.  HPOS is the default, so it should be for our tests too.
2. Disable the publishing of daily test results to the dashboard for non HPOS runs (they aren't being displayed there)

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Actions > Smoke test daily
3. Run the workflow using this branch and make sure there's no issues
4. Go to Actions > Run daily tests in an environment with HPOS enabled
5. Run this workflow using this branch and make sure there's no issues

Here's a couple if you don't want to wait:

- [HPOS disabled](https://github.com/woocommerce/woocommerce/actions/runs/7891861581)
- [Normal daily](https://github.com/woocommerce/woocommerce/actions/runs/7891865540)

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: .github/workflows/smoke-test-daily.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
             - name: Load docker images and start containers.
               working-directory: plugins/woocommerce
               env:
-                  ENABLE_HPOS: 0
+                  ENABLE_HPOS: 1
```
Comment (adimoldovan) [inline .github/workflows/smoke-test-daily.yml:42]: If HPOS enabled is default, do we still need this environment variable?

**Review by adimoldovan — COMMENTED** [review]
The change looks good to me, but I wonder if it would make sense to also rename the workflow to `non-hpos-build-and-e2e-tests-daily.yml`? The same with the job names in the workflow, a rename from `hpos-...` to `non-hpos...`?

File: .github/workflows/smoke-test-daily.yml (hunk 1 above)
Comment (lanej0) [reply .github/workflows/smoke-test-daily.yml:42]: Probably not, I just wanted to be explicit about it.

**Comment by lanej0** [comment]
> The change looks good to me, but I wonder if it would make sense to also rename the workflow to `non-hpos-build-and-e2e-tests-daily.yml`? The same with the job names in the workflow, a rename from `hpos-...` to `non-hpos...`?

Good call.  I made the update to the filename.

**Review by adimoldovan — APPROVED** [review]

**Pull request merged**